src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java

```diff
@@ -199,9 +204,11 @@ private void visitReturn(DetailAST ast) {
         // we can't identify which max to use for lambdas, so we can only assign
         // after the first return statement is seen
         if (ast.getFirstChild().getType() == TokenTypes.SEMI) {
+            context.hasVoidReturn = true;
             context.visitLiteralReturn(maxForVoid);
```

Comment:
usually accessing a field directly is frowned upon.
We have the `visit` method call right below the `hasVoidReturn`, so we should pass the value to that method instead.

Reply:
done